tests/testthat/test-printing.R

```diff
@@ -298,16 +298,18 @@ test_that("Inset works for table, ref_footnotes, and main footer", {
 
   # Building the table and trimming NAs
   tt <- build_table(lyt, DM)
-  tt <- trim_rows(tt)
+  tt <- prune_table(tt)
+  # tt <- trim_rows(tt)
 
   # Adding references
   # row_paths(tt)
   # row_paths_summary(tt)
+  # col_paths(tt)
+  # col_paths_summary(tt)
   txt1 <- "Not the best but very long one, probably longer than possible."
   txt2 <- "Why trimming does not take it out?"
   fnotes_at_path(tt, rowpath = c("SEX", "F", "AGE", "Mean")) <- txt1
-  fnotes_at_path(tt, rowpath = c("SEX", "UNDIFFERENTIATED")) <- txt2
```

Comment:
This should work as is why was it removed? If it doesn't work thats a bug

Reply:
So here, even with the current fix the two calls, `trim_rows(tt)` and `prune_table(tt)`, do not give the same results. The difference is that the empty nesting levels are kept with `trim_rows` while pruning takes them out. For example from #476 the result keeps UNDIFFERENTIATED in one case while the other takes it out:
```
                     all obs
                     (N=356)
  ——————————————————————————
  F                         
    Mean              34.13 
  M                         
    Mean              34.32 
  U                         
  UNDIFFERENTIATED      
```

About this, I heard @BFalquet saying that sometimes `trim_rows` is preferred because it keeps more structure that then can be more easily cbinded. Let me know if you want me to file an issue for having these be the same or if I should only test these two different behaviors.